File: task6.4.5.js
const xmlData = `
<list>
  <student>
    <name lang="en">
      <first>Ivan</first>
      <second>Ivanov</second>
    </name>
    <age>35</age>
    <prof>teacher</prof>
  </student>
  <student>
    <name lang="ru">
      <first>Петр</first>
      <second>Петров</second>
    </name>
    <age>58</age>
    <prof>driver</prof>
  </student>
</list>
`;

// Парсинг XML
const parser = new DOMParser();
const xmlDoc = parser.parseFromString(xmlData, "text/xml");

// Преобразование в объект
const result = { list: [] };
const students = xmlDoc.getElementsByTagName('student');

for (let i = 0; i < students.length; i++) {
  const student = students[i];
  const nameNode = student.getElementsByTagName('name')[0];
  const firstName = nameNode.getElementsByTagName('first')[0].textContent;
  const secondName = nameNode.getElementsByTagName('second')[0].textContent;
  const fullName = `${firstName} ${secondName}`;
  const lang = nameNode.getAttribute('lang');
  const age = parseInt(student.getElementsByTagName('age')[0].textContent, 10);
  const prof = student.getElementsByTagName('prof')[0].textContent;

  result.list.push({
    name: fullName,
    age: age,
    prof: prof,
    lang: lang
  });
}

// Вывод результата
console.log(result);

File: task6.4.5.py
import xml.etree.ElementTree as ET

xml_data = """
<list>
  <student>
    <name lang="en">
      <first>Ivan</first>
      <second>Ivanov</second>
    </name>
    <age>35</age>
    <prof>teacher</prof>
  </student>
  <student>
    <name lang="ru">
      <first>Петр</first>
      <second>Петров</second>
    </name>
    <age>58</age>
    <prof>driver</prof>
  </student>
</list>
"""

# Парсинг XML
root = ET.fromstring(xml_data)

# Преобразование в словарь
result = {"list": []}
for student in root.findall('student'):
    name = student.find('name')
    first_name = name.find('first').text
    second_name = name.find('second').text
    full_name = f"{first_name} {second_name}"
    lang = name.get('lang')
    age = int(student.find('age').text)
    prof = student.find('prof').text
    
    result["list"].append({
        "name": full_name,
        "age": age,
        "prof": prof,
        "lang": lang
    })

# Вывод результата
print(result)